plugins/auth-backend/src/providers/types.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import express from 'express';
+import passport from 'passport';
+
+export interface AuthProviderRouteHandlers {
+  start(
+    req: express.Request,
+    res: express.Response,
+    next: express.NextFunction,
+  ): Promise<any>;
+  frameHandler(
+    req: express.Request,
+    res: express.Response,
+    next: express.NextFunction,
+  ): express.Response<any>;
+  refresh?(
+    req: express.Request,
+    res: express.Response,
+    next: express.NextFunction,
+  ): Promise<any>;
+  logout(
+    req: express.Request,
+    res: express.Response,
+    next: express.NextFunction,
+  ): express.Response<any>;
+}
+export interface AuthProvider {
+  strategy(): passport.Strategy;
+  router?(): express.Router;
```

Comment:
will it be a normal thing to not have a router?

Reply:
All providers are based on passportJS and we add a default router (default routes with handler methods provided by the provider class).

Just keeping it open in the interface here so that a custom provider in the future could supply its own router.